### PR TITLE
Issue PS-1497 feat: optimize learner, facilitator, team leader  create updatemodal

### PR DESCRIPTION
**Pull request by AkshataKatwal16**

### CODE REVIEW

File: src/components/CommonUserModal.tsx [hunk 1]
```diff
@@ -145,8 +161,8 @@ const AddLearnerModal: React.FC<AddLearnerModalProps> = ({
       tenantCohortRoleMapping: [
         {
           tenantId: tenantId,
-          roleId: RoleId.STUDENT,
-          cohortId: [selectedCenterCode],
+          roleId:userType==="STUDENT"? RoleId.STUDENT: userType==="TEACHER"?RoleId.TEACHER:RoleId.TEAM_LEADER,
```
Comment (itsvick) [inline src/components/CommonUserModal.tsx:164]: @AkshataKatwal16 get the types from the app.constant files

File: src/components/CommonUserModal.tsx [hunk 2]
```diff
@@ -307,7 +347,7 @@ const AddLearnerModal: React.FC<AddLearnerModalProps> = ({
         open={open}
         onClose={onClose}
         showFooter={false}
-        modalTitle={!isEditModal?t("LEARNERS.NEW_LEARNER"): t("LEARNERS.EDIT_LEARNER")}
+        modalTitle={!isEditModal? userType === "STUDENT"?t("LEARNERS.NEW_LEARNER"):userType === "TEACHER"?t("FACILITATORS.NEW_FACILITATOR"):t("TEAM_LEADERS.NEW_TEAM_LEADER"): userType === "STUDENT"?t("LEARNERS.EDIT_LEARNER"):userType === "TEACHER"?t("FACILITATORS.EDIT_FACILITATOR"):t("TEAM_LEADERS.EDIT_TEAM_LEADER")}
```
Comment (itsvick) [inline src/components/CommonUserModal.tsx:350]: @AkshataKatwal16 use app.constant file for constants and try to move this logic to top of the file

File: src/components/CommonUserModal.tsx [hunk 3]
```diff
@@ -335,7 +375,7 @@ const AddLearnerModal: React.FC<AddLearnerModalProps> = ({
               handleBlockChangeWrapper={handleBlockChangeWrapper}
               isMobile={isMobile}
               isMediumScreen={isMediumScreen}
-              isCenterSelection={true}
+              isCenterSelection={userType!=="TEAM LEADER"?true: false}
```
Comment (itsvick) [inline src/components/CommonUserModal.tsx:378]: @AkshataKatwal16 take it from the app.constant

**Review by itsvick — CHANGES_REQUESTED** [review]

File: src/components/CommonUserModal.tsx (hunk 3 above)
Comment (AkshataKatwal16) [reply src/components/CommonUserModal.tsx:379]: done

File: src/components/CommonUserModal.tsx (hunk 2 above)
Comment (AkshataKatwal16) [reply src/components/CommonUserModal.tsx]: done

File: src/components/CommonUserModal.tsx (hunk 1 above)
Comment (AkshataKatwal16) [reply src/components/CommonUserModal.tsx]: done

**Pull request merged**